File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EmptyMethod.OverrideVirtual.vb
Imports System
Imports System.Diagnostics
Imports System.Runtime.InteropServices

Namespace Tests.Diagnostics
    Public Class FooBase
        Public Overridable Sub Method()
        End Sub
    End Class

    Public Class FooImpl
        Inherits FooBase

        Public Overrides Sub Method() ' Noncompliant
        End Sub
    End Class
End Namespace

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EmptyMethod.OverrideVirtual.cs
using System;
using System.Diagnostics;
using System.Runtime.InteropServices;

namespace Tests.Diagnostics
{
    public class FooBase
    {
        public virtual void Method()
        {

        }
    }

    public class FooImpl: FooBase
    {
        public override void Method() // Noncompliant
        {

        }
    }
}
